File: hostera24/lib/utils/media_url.dart
import 'package:hostera24/config/api_config.dart';

/// Logo-uri salvate pe server folosesc /uploads/... — aliniem host-ul la API_BASE_URL.
String? resolveUploadsMediaUrl(String? url) {
  if (url == null || url.trim().isEmpty) return null;

  final trimmed = url.trim();
  final idx = trimmed.indexOf('/uploads/');
  if (idx < 0) return trimmed;

  final suffix = trimmed.substring(idx + '/uploads'.length);
  final base = ApiConfig.baseUrl.replaceAll(RegExp(r'/+$'), '');
  return '$base/uploads$suffix';
}

File: frontend-hostera24/lib/media-url.ts
import { apiBaseUrlFromEnv } from "@/lib/env";

/** Aliniază URL logo /uploads la API public (fix host vechi din DB). */
export function resolveUploadsMediaUrl(
  url: string | null | undefined,
): string | null {
  if (!url?.trim()) return null;

  const trimmed = url.trim();
  const idx = trimmed.indexOf("/uploads/");
  if (idx < 0) return trimmed;

  const suffix = trimmed.slice(idx + "/uploads".length);
  const api = apiBaseUrlFromEnv();

  return `${api}/uploads${suffix}`;
}
